src/index.js

```diff
@@ -239,9 +248,13 @@ export default class RNPickerSelect extends PureComponent {
 
             // If TextInput is below picker modal, scroll up
             if (textInputBottomY > modalY) {
-                this.props.scrollViewRef.current.scrollTo({
-                    y: textInputBottomY - modalY + this.props.scrollViewContentOffsetY,
-                });
+                // Wait until the modal animation finishes, so the scrolling is effective when PickerAvoidingView is
+                // used
+                setTimeout(() => {
```

Comment:
@Julesssss

> Can we figure out a way of avoiding the setTimeout? Possible via a hook?

Reply:
Are you concerned about the component being unmounted before the timeout finishes?

Reply:
We just try to avoid using `setTimeout` as a pattern if possible

https://github.com/Expensify/App/blob/ab9b7d9ef1fb1656f5ccfc28ca4a68820fa373b4/contributingGuides/PR_REVIEW_GUIDELINES.md?plain=1#L56

Reply:
Ok, so [this commit](https://github.com/Expensify/react-native-picker-select/pull/8/commits/eeb3bff23ef3d060148057263466aa674a5fadcc) wraps that timeout in a hook, preventing it from occurring when the component is unmounted.

The thing is that we can't scroll until the layout hasn't been updated, and the layout isn't updated until the animation finishes, not to cause a "blink" (as [requested here](https://github.com/Expensify/App/pull/17279#discussion_r1164239617) by @0xmiroslav).

>  Make sure any `setTimeout`s are absolutely necessary, not just a workaround for a situation that isn’t fully understood (e.g. a race condition).

The second part is definitely not the case here. The situation is fully understood. The first part is arguable. It's not **absolutely** necessary to use a time-based solution here, but creating a communication channel between `PickerAvoidingView` and the picker itself would be quite a big effort here.

Both `PickerAvoidingView` and `scrollToInput` share the `IOS_MODAL_ANIMATION_DURATION_MS` constant, so their cross-dependence is clearly marked in the code.

Reply:
@Julesssss What's your perspective on this? I tried to explain why I believe that a timer-based solution does not indicate a non-fully understood situation here. While the hook-based solution doesn't remove the timer completely, it improves how the timer is managed, by coupling it to the component lifecycle (canceling on umount). Does this look acceptable?

I believe that this is the only real topic left in this pair of PRs. It really would be great to start closing this, I think!

Reply:
Honestly I'm not 100% sure what's best here, but you make a good case. What do you think @0xmiroslav ?

Reply:
@0xmiroslav writes [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1681852803227919?thread_ts=1681386144.253199&cid=C01GTK53T8Q):

> If timeout removal is that complex and no other solution, then fine

I would say that there is no simple way to drop the timeout-based approach here. I'll resolve the thread.